### PR TITLE
Return proper error when backups are empty for details command

### PR DESCRIPTION
**Pull request by meain**
---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by ntolia** [comment]
Can we add a Changelog entry for this, please?

File: src/cli/backup/backup.go [hunk 1]
```diff
@@ -394,6 +394,10 @@ func genericDetailsCore(
 		return nil, clues.Wrap(errs.Failure(), "Failed to get backup details in the repository")
 	}
 
+	if len(d.Entries) == 0 {
+		return nil, clues.New("empty backup")
```
Comment (ryanfkeepers) [inline src/cli/backup/backup.go:398]: nit, since "empty" can be interpreted many ways.
```suggestion
		return nil, clues.New("no items in backup")
```

**Comment by meain** [comment]
Updated the message and added a changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased] (beta)
 
+### Changed
+- If we try to see details of an empty backup, it now returns a more helpful error message.
```
Comment (ryanfkeepers) [inline CHANGELOG.md:11]:
```suggestion
- Running `backup details` on an empty backup returns a more helpful error message.
```

**Review by ryanfkeepers — APPROVED** [review]

File: src/cli/backup/backup_test.go [hunk 3]
```diff
@@ -66,3 +68,30 @@ func (suite *BackupUnitSuite) TestGenericDetailsCore() {
...
+	sel.Include(sel.AllData())
+
+	_, err := genericDetailsCore(
+		ctx,
+		bg,
+		"backup-ID",
+		sel.Selector,
+		control.DefaultOptions())
+	require.Error(t, err, "has error")
+	assert.ErrorContains(t, err, "no items in backup", clues.ToCore(err))
```
Comment (ryanfkeepers) [inline src/cli/backup/backup_test.go:96]: minor: error message checks are brittle.  An error sentinel and ErrorIs check would be more resilient.

**Pull request merged**